### PR TITLE
TER-94 add ABS stub loader

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: TER-94
driver-common: https://github.com/keboola/php-storage-driver-common/pull/33

### CODE REVIEW

File: tests/functional/StorageHelper/StorageABSTrait.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        string $accountName,
+        string $accountKey,
+        string $container,
+        string $permissions = 'rwl'
+    ): string {
+        $sasHelper = new BlobSharedAccessSignatureHelper(
+            $accountName,
+            $accountKey,
+        );
+        $expirationDate = (new DateTime())->modify('+1hour');
```
Comment (jirkasemmler) [inline tests/functional/StorageHelper/StorageABSTrait.php:69]: staci ta 1h?

File: tests/functional/UseCase/Table/ImportTableFromFileTest.php [hunk 2]
```diff
@@ -417,29 +372,19 @@ public function testImportTableFromTableFullLoadSlicedWithoutDeduplication(
...
         if ($compression === TableImportFromFileCommand\CsvTypeOptions\Compression::GZIP) {
-            $cmd->setFilePath(
-                (new FilePath())
-                    ->setRoot((string) getenv('AWS_S3_BUCKET'))
-                    ->setPath('sliced/accounts-gzip')
-                    ->setFileName('S3.accounts-gzip.csvmanifest')
+            $this->setFilePathAndCredentials(
+                $cmd,
+                'sliced/accounts-gzip',
+                '%MANIFEST_PREFIX%accounts-gzip.csvmanifest',
```
Comment (jirkasemmler) [inline tests/functional/UseCase/Table/ImportTableFromFileTest.php:378]: tak nejsem z toho nadsenej, ale ok

File: composer.json [hunk 3]
```diff
@@ -50,18 +50,28 @@
         "lock": false
     },
     "scripts": {
-        "protobufTests": "rm -rf ./tests/generated/* && protoc $(find tests/proto -iname \"*.proto\") --php_out=tests/generated",
         "phplint": "parallel-lint -j 10 --exclude vendor --exclude bin .",
         "phpcs": "phpcs --extensions=php tests --ignore=./tests/generated/*",
         "phpcbf": "phpcbf --extensions=php tests --ignore=./tests/generated/*",
         "phpstan": "phpstan analyse --no-progress",
-        "phpunit": "rm -rf ./tests/generated/* && protoc $(find tests/proto -iname \"*.proto\") --php_out=tests/generated && phpunit",
+        "protobufTests": "rm -rf ./tests/generated/* && protoc $(find tests/proto -iname \"*.proto\") --php_out=tests/generated",
```
[the remark this answers is not in the document]
Comment (martinjunger) [reply composer.json:57]: jo, to je tam proto, aby to jen vygenerovalo ty `proto` soubory

File: tests/functional/StorageHelper/StorageABSTrait.php (hunk 1 above)
Comment (martinjunger) [reply tests/functional/StorageHelper/StorageABSTrait.php:69]: to je prevzaty z IE libky - rekl bych ze jo, je to pro jeden beh 1 test casu

**Comment by martinjunger** [comment]
S3: ImportTableFromTableTest::testImportOfBigTableToSmallBucket - pada na
> Doctrine\DBAL\Exception\DriverException: An exception occurred while executing a query: [Teradata][ODBC Teradata Driver][Teradata Database](-3561)Cannot DROP USER 04904FD159E404EFF3C699A9F716C421129B943DAF45F348B3126D8A9AE7B780_USER while the user is logged on.

Ale lokalne prochazi:
php-storage-driver-teradata – BaseCase php 2023-01-27 16-31-53

**Comment by martinjunger** [comment]
ABS: testy jsem fixnul, pripadne upravil (export slice do F0000x)

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
cajk, az to projde, tak to prosim jen trochu posquashuj at to vypada trochu lip :) pak to bude ready to merge

**Review by jirkasemmler — APPROVED** [review]

**Pull request merged**